ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala

```diff
@@ -95,12 +95,39 @@ trait WriteService
     * @param estimatedInterpretationCost Estimated cost of interpretation that may be used for
     *                                    handling submitted transactions differently.
     * @param telemetryContext            Implicit context for tracing.
-    * @return an async result of a SubmissionResult
     */
   def submitTransaction(
       submitterInfo: SubmitterInfo,
       transactionMeta: TransactionMeta,
       transaction: SubmittedTransaction,
       estimatedInterpretationCost: Long,
   )(implicit telemetryContext: TelemetryContext): CompletionStage[SubmissionResult]
+
+  /** Record the rejection of a command submission on the ledger.
+    *
+    * This method must be thread-safe.
+    *
+    * This method is used by the Ledger API server to deliver interpretation failures as definitive answers to the
+    * Ledger API client via completions; thereby allowing the Ledger API client to safely decide to not retry
+    * the command if desired.
+    *
+    * The result is communicated asynchronously via a [[ReadService]] implementation backed by the same participant
+    * state as this [[WriteService]]. Successful recording is communicated using a [[Update.CommandRejected]]
+    * with [[SubmitterInfo]] and [[Update.CommandRejected.definiteAnswer]].
+    * If the recording as a rejection fails (e.g., due to deduplication or violations of the submission rank),
+    * the failure should be communicated using a [[Update.CommandRejected]] with [[SubmitterInfo]]
+    * and not [[Update.CommandRejected.definiteAnswer]].
+    *
+    * Recorded rejections fall under the deduplication and submission rank guarantees
+    * described in [[ReadService.stateUpdates]].
+    *
+    * @param submitterInfo the information provided by the submitter for correlating this submission
+    *                      with its rejection on the associated [[ReadService]].
+    * @param reason The rejection reason to be included in the [[Update.CommandRejected]]
+    * @param telemetryContext Implicit context for tracing.
+    */
+  def rejectSubmission(
+      submitterInfo: SubmitterInfo,
```

Comment:
```suggestion
      submitterInfo: SubmitterInfo,
      submissionTime : Timestamp,
      submissionConfig: LedgerConfig,
```

Reply:
Group into `SubmissionContext` case class and add to `submitTransactoin` as well